### PR TITLE
refactor(verify): pull out a shared component so it can be reused with CAB

### PR DESCRIPTION
**Pull request by satish-ravi**
### Description

Pulls out parts of the PhoneVerificationCodeInputScreen so it can be reused with the phone verification for cloud backup. Kept the help header outside because it's part of the nav header and the CTA on the bottom sheet is different

### Test plan

Unit tests, manually. Left the unit tests for PhoneVerificationCodeInputScreen as is since they test the full verification flow with the hook.

### Related issues

- To reuse for act-759

### Backwards compatibility

Yes

### CODE REVIEW

File: src/verify/VerificationCodeInputWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+interface Props {
+  phoneNumber: string
+  verificationStatus: PhoneNumberVerificationStatus
+  setSmsCode: (code: string) => void
+  onResendSms?: () => void
+  onSuccess: () => void
+}
+
+function VerificationCodeInputWrapper({
```
Comment (kathaypacific) [inline src/verify/VerificationCodeInputWrapper.tsx:22]:
```suggestion
function VerificationCodeInput({
```

File: src/verify/VerificationCodeInputWrapper.test.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        verificationStatus={verificationStatus}
+        onSuccess={mockOnSuccess}
+        onResendSms={showResend ? mockOnResendSms : undefined}
+        setSmsCode={mockSetSmsCode}
+      />
+    )
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    jest.useFakeTimers()
```
Comment (kathaypacific) [inline src/verify/VerificationCodeInputWrapper.test.tsx:32]: nit: i think this is already the default defined in jest_setup?

**Review by kathaypacific — APPROVED** [review]
👨🏼‍🎨

File: src/verify/VerificationCodeInputWrapper.test.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      <VerificationCodeInputWrapper
+        phoneNumber={mockPhoneNumber}
+        verificationStatus={verificationStatus}
+        onSuccess={mockOnSuccess}
+        onResendSms={showResend ? mockOnResendSms : undefined}
+        setSmsCode={mockSetSmsCode}
+      />
+    )
+
+  beforeEach(() => {
```
Comment (jophish) [inline src/verify/VerificationCodeInputWrapper.test.tsx:30]: Can we add a test to ensure the paste button appears if the clipboard contents are set appropriately (and that it doesn't appear if the contents are not valid)? We should be able to check the testID from [here](https://github.com/valora-inc/wallet/blob/main/src/components/ClipboardAwarePasteButton.tsx#L40).

**Review by jophish — APPROVED** [review]
Nice abstraction -- one comment about a potential test to add.

**Review by jophish — APPROVED** [review]
:tada:

**Pull request merged**